packages/runtime-dom/src/index.ts

```diff
@@ -103,14 +105,25 @@ function injectNativeTagCheck(app: App) {
   })
 }
 
-function normalizeContainer(container: Element | string): Element | null {
+function normalizeContainer(
+  container: Element | ShadowRoot | string
+): Element & { __isShadowRoot?: boolean } | null {
   if (isString(container)) {
     const res = document.querySelector(container)
     if (__DEV__ && !res) {
       warn(`Failed to mount app: mount target selector returned null.`)
     }
     return res
   }
+  if (container instanceof ShadowRoot) {
```

Comment:
`mode: 'closed'` doesn't actually prevent Vue from mounting to it (see https://blog.revillweb.com/open-vs-closed-shadow-dom-9f3d7427d1af) so this whole check is unnecessary

Reply:
Isn't it working? https://codepen.io/yyx990803/pen/KKMgooN?editors=1010

Reply:
I know why now. It's a regression of Vimium, an extension that I enabled in my browsers. https://github.com/philc/vimium/issues/853

Reply:
So I wonder if warnings should be retained so that developers can be aware that some of the problems may be caused by the closed shadow root, for what the user side is using is beyond developers' control